src/components/AddressSearch/index.tsx

```diff
@@ -1,184 +1,77 @@
-    maxInputLength: PropTypes.number,
-
-    /** The result types to return from the Google Places Autocomplete request */
-    resultTypes: PropTypes.string,
-
-    /** Information about the network */
-    network: networkPropTypes.isRequired,
-
-    /** Location bias for querying search results. */
-    locationBias: PropTypes.string,
-
-    ...withLocalizePropTypes,
-};
-
-const defaultProps = {
-    inputID: undefined,
-    shouldSaveDraft: false,
-    onBlur: () => {},
-    onPress: () => {},
-    errorText: '',
-    hint: '',
-    value: undefined,
-    defaultValue: undefined,
-    containerStyles: [],
-    isLimitedToUSA: false,
-    canUseCurrentLocation: false,
-    renamedInputKeys: {
-        street: 'addressStreet',
-        street2: 'addressStreet2',
-        city: 'addressCity',
-        state: 'addressState',
-        zipCode: 'addressZipCode',
-        lat: 'addressLat',
-        lng: 'addressLng',
-    },
-    maxInputLength: undefined,
-    predefinedPlaces: [],
-    resultTypes: 'address',
-    locationBias: undefined,
-};
-
-function AddressSearch({
-    canUseCurrentLocation,
-    containerStyles,
-    defaultValue,
-    errorText,
-    hint,
-    innerRef,
-    inputID,
-    isLimitedToUSA,
-    label,
-    maxInputLength,
-    network,
-    onBlur,
-    onInputChange,
-    onPress,
-    predefinedPlaces,
-    preferredLocale,
-    renamedInputKeys,
-    resultTypes,
-    shouldSaveDraft,
-    translate,
-    value,
-    locationBias,
-}) {
+function AddressSearch(
+    {
+        canUseCurrentLocation = false,
+        containerStyles,
+        defaultValue,
+        errorText = '',
+        hint = '',
+        inputID,
+        isLimitedToUSA = false,
+        label,
+        maxInputLength,
+        onBlur,
+        onInputChange,
+        onPress,
+        predefinedPlaces = [],
+        preferredLocale,
+        renamedInputKeys = {
+            street: 'addressStreet',
+            street2: 'addressStreet2',
+            city: 'addressCity',
+            state: 'addressState',
+            zipCode: 'addressZipCode',
+            lat: 'addressLat',
+            lng: 'addressLng',
+        },
+        resultTypes = 'address',
+        shouldSaveDraft = false,
+        value,
+        locationBias,
+    }: AddressSearchProps,
+    ref: ForwardedRef<HTMLElement>,
+) {
     const theme = useTheme();
     const styles = useThemeStyles();
     const StyleUtils = useStyleUtils();
+    const {translate} = useLocalize();
+    const {isOffline} = useNetwork();
     const [displayListViewBorder, setDisplayListViewBorder] = useState(false);
     const [isTyping, setIsTyping] = useState(false);
     const [isFocused, setIsFocused] = useState(false);
-    const [searchValue, setSearchValue] = useState(value || defaultValue || '');
-    const [locationErrorCode, setLocationErrorCode] = useState(null);
+    const [searchValue, setSearchValue] = useState(value ?? defaultValue ?? '');
```

Comment:
Is it possible that `value` will be `''` and this way an empty string instead of default value will be used? If so it will be better to use `||` as it initially was.

Reply:
Right, changed